### ISSUE TITLE
Possible LINQ to SQL conversion problem using Contains on string property

### ISSUE THREAD

**Issue by nicholaslambertucci-ett** [SapientGuardian/SapientGuardian.EntityFrameworkCore.MySql#15]
Hi everyone, 
I encountered an MySqlException when I tried to query a mapped entity using Contains method on a string property. Below the code used to perform the query:

```
var users = await this._dbContext.Set<FrontEndUser>()
     .Where(u => u.Surname.Contains("pp"))
    .Skip(skip)
    .Take(take)
    .ToArrayAsync();
```

The exception message was:

> You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '+ '%'
> ORDER BY `u`.`id`
>  LIMIT 10 OFFSET 0'

With the help of a profiler I've extracted the query that throw the exception:

> SELECT `u`.`id`, `u`.`consentAgreedAt`, `u`.`email`, `u`.`firstname`, `u`.`modifiedAt`, `u`.`nickname`, `u`.`password`, `u`.`surname`, `u`.`username`, `u`.`type`
> FROM `Users` AS `u`
> WHERE (`u`.`type` = 0) AND `u`.`surname` LIKE ('%' + 'pp') + '%'
> LIMIT 10 OFFSET 0

The where condition _`u`.`surname` LIKE ('%' + 'pp') + '%'_  is causing the SYNTAX ERROR.

Have you ever encountered this issue?

**Comment by nicholaslambertucci-ett** [comment]
Hi everyone, 
I've created the pull request #16 that contains a possible solution to the issue described in this topic.

**Comment by SapientGuardian** [comment]
@how02 any thoughts on this issue, or on the PR?

**Comment by how02** [comment]
The PR indeed fix the issue. Nice work :+1:

**Comment by SapientGuardian** [comment]
Your fix is out in 7.1.6, thanks for your contribution!
